lib/core/storage.js

```diff
@@ -128,79 +132,101 @@ export default class Storage {
     if (typeof localStorage === 'undefined' || !this.options.localStorage) {
       return
     }
 
     const _key = this.options.localStorage.prefix + key
 
-    if (isUnset(value)) {
-      localStorage.removeItem(_key)
-    } else {
-      localStorage.setItem(_key, isJson ? JSON.stringify(value) : value)
+    try {
+      if (isObjectOrArray(value)) {
+        localStorage.setItem(_key, JSON.stringify(value))
+      } else {
+        localStorage.setItem(_key, value)
+      }
+    } catch (e) {
+      if (!this.options.ignoreExceptions) {
+        throw e
+      }
     }
 
     return value
   }
 
-  getLocalStorage (key, isJson) {
+  getLocalStorage (key) {
     if (typeof localStorage === 'undefined' || !this.options.localStorage) {
       return
     }
 
     const _key = this.options.localStorage.prefix + key
 
     const value = localStorage.getItem(_key)
-    if (value === 'false') {
-      return false
-    }
 
-    return isJson ? JSON.parse(value) : value
+    return decodeValue(value)
+  }
+
+  removeLocalStorage (key) {
+    const _key = this.options.localStorage.prefix + key
+    localStorage.removeItem(_key)
   }
 
   // ------------------------------------
   // Cookies
   // ------------------------------------
+  getCookies () {
+    const cookieStr = process.client
+      ? document.cookie
+      : this.ctx.req.headers.cookie
+
+    return parseCookie(cookieStr || '') || {}
+  }
 
   setCookie (key, value, options = {}) {
-    if (process.server || !this.options.cookie) {
+    if (!this.options.cookie) {
       return
     }
 
     const _key = this.options.cookie.prefix + key
-
     const _options = Object.assign({}, this.options.cookie.options, options)
+    const _value = encodeValue(value)
 
-    if (isUnset(value)) {
-      Cookies.remove(_key, _options)
-    } else {
-      Cookies.set(_key, value, _options)
+    const serializedCookie = serializeCookie(_key, _value, _options)
+
+    if (process.client) {
+      // Set in browser
+      document.cookie = serializedCookie
+    } else if (process.server && this.ctx.res) {
+      // Send Set-Cookie header from server side
+      this.ctx.res.setHeader('Set-Cookie', serializedCookie)
     }
 
     return value
   }
 
-  getCookie (key, isJson) {
+  getCookie (key) {
     if (!this.options.cookie || (process.server && !this.ctx.req)) {
       return
     }
 
     const _key = this.options.cookie.prefix + key
 
-    const cookieStr = process.client
-      ? document.cookie
-      : this.ctx.req.headers.cookie
+    const cookies = this.getCookies()
 
-    const cookies = parseCookie(cookieStr || '') || {}
-    const value = cookies[_key]
-    if (value === 'false') {
-      return false
-    }
-    return isJson ? JSON.parse(value) : value
+    const value = cookies[_key] ? decodeURIComponent(cookies[_key]) : undefined
+
+    return decodeValue(value)
+  }
+
+  removeCookie (key, options) {
+    this.setCookie(key, undefined, Object.assign({ maxAge: -1 }, options))
```

Comment:
What about changes in the rest of the auth module? For instance - the new removeUniversal helper should now be called from [this line in auth.js](https://github.com/nuxt-community/auth-module/blob/dev/lib/core/auth.js#L350). If not - the cookie will not be removed, but simply set to "null", without expiration date.

Reply:
Will update other areas as well. Thanks for the tip. (however, I think we should bump to 5.x after this PR as this is a breaking change)

Reply:
(for now, added backward compatible conditions to not be a breaking change)